docs/site/guide/overview.md

```diff
@@ -0,0 +1,134 @@
+
+Swift for TensorFlow is a next-generation platform for machine learning,
+incorporating the latest research across machine learning, compilers,
+differentiable programming, systems design, and beyond. This is an early-stage
+project: it is not feature-complete nor production-ready, but it is ready for
+_pioneers_ to try in projects, give feedback, and help shape the future!
+
+The Swift for TensorFlow project is currently focusing on 2 kinds of users:
+
+1.  **Advanced ML researchers** who are limited by current ML frameworks. Swift
+    for TensorFlow's advantages include seamless integration with a modern
+    general-purpose language, allowing for more dynamic and sophisticated
+    models. Fast abstractions can be developed in "user-space" (as opposed to in
+    C/C++, aka "framework-space"), resulting in modular APIs that can be easily
+    customized.
+
+2.  **ML learners** who are just getting started with machine learning. Thanks
+    to Swift's support for quality tooling (e.g. context-aware autocompletion),
+    Swift for TensorFlow can be one of the most productive ways to start
+    learning the fundamentals of machine learning.
+
+## Swift
+
+Swift is an [open source](https://swift.org/) general-purpose programming language, which has a 
+large and growing user base.  We chose Swift because it has an
+[open language design process](https://github.com/apple/swift-evolution) and for specific technical 
+reasons detailed in the 
+"[Why *Swift* for TensorFlow](https://github.com/BradLarson/swift/blob/main/docs/WhySwiftForTensorFlow.md)"
+document.  We assume that most readers are unfamiliar with it, so we’ll briefly touch on some 
+additional important things about it here.
+
+The development of Swift started in 2010, and aimed to bring the best practices in programming 
+language design together into one system rather than trying for academic novelty or to religiously 
+propagate programming methodologies.  As a result, it supports multi-paradigm development (e.g. 
+functional, OOP, generic, procedural, etc) all in one system,  and brings many well-known concepts 
+from academic languages (e.g.
+[pattern matching](http://alisoftware.github.io/swift/pattern-matching/2016/03/27/pattern-matching-1/), 
+[algebraic data types](https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/Enumerations.html),
+and type classes) into the forefront.  Instead of strongly encouraging developers to rewrite all 
+their code in Swift, it pragmatically focuses on interoperability with other languages, e.g., 
+allowing you to directly import C header files and use them without an 
+[FFI](https://en.wikipedia.org/wiki/Foreign_function_interface) and (now) the ability to use Python 
+APIs without wrappers.
+
+Swift has the audacious goal of spanning all the way from low-level systems programming to 
+high-level scripting, with a focus on being
+[easy to learn and use](https://www.apple.com/swift/playgrounds/). 
```

Comment:
May I recommend a quote from Chris (@lattner) from a talk (don't remember which one): "Swift was conceived as a full-stack programming language".

I'd also recommend mentioning type safety as a core feature (looking at my notes: Swift is "compiler-friendly and designed to be fast from the beginning.") Would it also make sense to emphasize that Swift is fast and human-friendly? I see "fast" being mentioned once here already though.

I've also got a quote from Jeremy H in my notes: _"...the first serious effort I’ve seen to incorporate differentiable programming deep in to the heart of a widely used language that is designed from the ground up for performance."_ I see "differentiable" is mentioned several time already and maybe it can make sense to add the "effort... to incorporate... in to the ... language" 🤷